exe/pyinstaller/aws.spec

```diff
@@ -13,6 +13,35 @@ aws_a = Analysis(['../../bin/aws'],
              win_no_prefer_redirects=False,
              win_private_assemblies=False,
              cipher=block_cipher)
+
+# Replace the Python.framework directory with a top-level Python executable
+# This is easier for our internal signing and notarization code,
+# since we're not signing via PyInstaller
+updated_binaries = []
+for dest, src, typecode in aws_a.binaries:
+    # Look for the actual Python executable, regardless of the version
+    if dest.startswith('Python.framework/Versions/') and dest.endswith('/Python'):
+        # and move it to the top
+        dest = 'Python'
+    updated_binaries.append((dest, src, typecode))
+aws_a.binaries = updated_binaries
+
+# Remove the symlinks and the Info.plist related to Python.framework
+# since we're using the top-level executable above
+updated_datas = []
+for dest, src, typecode in aws_a.datas:
+    if (dest.startswith('Python.framework/') or (dest == 'Python' and typecode == 'SYMLINK')):
+        continue
+    updated_datas.append((dest, src, typecode))
+aws_a.datas = updated_datas
+
+# Verify that there are no remaining symlinks
+for dest, src, typecode in aws_a.datas:
```

Comment:
Is there a reason we don't want to do this check for binaries?

Also, if a data file is a symlink but does not get picked up as an eligible/signable file by our build system, does it matter? Should we just let the signing job fail instead of trying to raise an exception in the build process?